pillar/common.sls

```diff
@@ -16,3 +18,13 @@ salt_minion:
       log_granular_levels:
         salt: warning
         salt.loader: warning
+    saltenvs:
+      {% if 'production' in environment %}
+      saltenv: production
+      lock_saltenv: True
+      pillarenv: production
+      {% else %}
+      saltenv: base
+      pillarenv_from_saltenv: True
+      {% endif %}
+      state_top_saltenv: base
```

Comment:
Not very clear on this, but since it's outside the conditional, would this not cause the base saltenv thus the master branch to apply and overwrite the one set for the production environment?

Reply:
So, that's actually for handling which top file gets used for targeting, not which state files/saltenv to use for rendering.